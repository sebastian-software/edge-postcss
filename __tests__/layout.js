import { compile } from "./core"

// ====================================================
// ================== LAYOUT ==========================
// ====================================================

test("Lost Grid", () =>
  compile(".grid { lost-column: 3/12 }")
)

test("Grid KISS", () =>
  compile(`
  .gridTest {
    grid-kiss:
      "+-------------------------------+      "
      "|           header ↑            | 120px"
      "+-------------------------------+      "
      "                                       "
      "+-- 30% ---+  +--- auto --------+      "
      "| .sidebar |  |       main      | auto "
      "+----------+  +-----------------+      "
      "                                       "
      "+-------------------------------+      "
      "|              ↓                | 60px "
      "|         → footer ←          |      "
      "+-------------------------------+      "
  }
`)
)

test("Flow Root", () =>
  compile(".selector { display: flow-root; }")
)
