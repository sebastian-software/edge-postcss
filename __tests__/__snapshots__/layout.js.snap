// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flow Root 1`] = `
".selector{
display:flow-root;
-webkit-column-count:1;
-moz-column-count:1;
column-count:1
}"
`;

exports[`Grid KISS 1`] = `
".gridTest{
display:grid;
-ms-flex-line-pack:justify;
align-content:space-between;
grid-template:\\"a a\\" 120px \\"b c \\" 1fr \\"e e\\" 60px/30% 1fr
}

.gridTest>header{
grid-area:a;
-ms-flex-item-align:start;
align-self:start
}

.gridTest>.sidebar{
grid-area:b
}

.gridTest>main{
grid-area:c
}

.gridTest>footer{
grid-area:e;
justify-self:center;
-ms-flex-item-align:end;
align-self:end
}

@supports not (grid-template-areas:\\"test\\"){
.gridTest{
position:relative;
display:block;
width:100%;
height:100%
}

.gridTest>*{
position:absolute;
box-sizing:border-box
}

.gridTest>header{
max-height:120px;
width:100%;
top:0;
left:0
}

.gridTest>.sidebar,.gridTest>main{
height:calc(100% - 180px);
width:30%;
top:120px;
left:0
}

.gridTest>main{
width:70%;
left:30%
}

.gridTest>footer{
transform:translateX(-50%);
max-height:60px;
max-width:100%;
bottom:0;
left:50%
}
}

@media screen and (min-width:0\\\\0){
.gridTest{
position:relative;
display:block;
width:100%;
height:100%
}

.gridTest>*{
position:absolute;
box-sizing:border-box
}

.gridTest>header{
max-height:120px;
width:100%;
top:0;
left:0
}

.gridTest>.sidebar,.gridTest>main{
height:calc(100% - 180px);
width:30%;
top:120px;
left:0
}

.gridTest>main{
width:70%;
left:30%
}

.gridTest>footer{
transform:translateX(-50%);
max-height:60px;
max-width:100%;
bottom:0;
left:50%
}
}"
`;

exports[`Lost Grid 1`] = `
".grid{
width:calc(24.975% - 22.5px)
}

.grid:nth-child(12n+1):after,.grid:nth-child(n):after{
content:'';
display:block;
clear:both
}

.grid:nth-child(n){
float:left;
margin-right:30px
}

.grid:last-child{
margin-right:0
}

.grid:nth-child(12n){
margin-right:0;
float:right
}"
`;
